### PR TITLE
Added `Remove-RegistryKey` to RegistryHelpers

### PR DESCRIPTION
**Pull request by sethbrasile**
And updated documentation for new function.

### CODE REVIEW

File: Function.Registry-Helpers.ps1 [hunk 1]
```diff
@@ -109,6 +109,20 @@ function Remove-RegistryValue {
...
+# Remove-RegistryKey deletes an entire key. This one requires `-Path` to be set and will not use pre-set $regPath
+function Remove-RegistryKey {
+    param(
+        [Parameter(Mandatory = $true)]
+        [string]$Path
+    )
+
+    If (!$Path) {
+        Throw 'Remove-RegistryKey could not continue. Path was not specified or was invalid! Please specify $Path to a valid Registry key path.'
+    }
```
Comment (wmmatt) [inline Function.Registry-Helpers.ps1:121]: Since you've made `$Path` Mandatory you'd never have a `$null` here ;)

**Review by wmmatt — CHANGES_REQUESTED** [review]
I think you can remove that if `$null` check, see whatcha think.

**Comment by sethbrasile** [comment]
Need to make sure input is actually a string, probably need to regex it to make sure it's a valid registry path.

Right now, this function could accidentally nuke a folder if accidentally fed a word that happens to be a folder in the current working directory.

File: Function.Registry-Helpers.ps1 (hunk 1 above)
Comment (sethbrasile) [reply Function.Registry-Helpers.ps1]: I believe this is handled by the more thorough check I added via regex. Ready for review again.

File: Function.Registry-Helpers.ps1 [hunk 2]
```diff
@@ -116,8 +116,12 @@ function Remove-RegistryKey {
         [string]$Path
     )
 
-    If (!$Path) {
-        Throw 'Remove-RegistryKey could not continue. Path was not specified or was invalid! Please specify $Path to a valid Registry key path.'
+    # currently only handles HKLM, need to adjust regex if you want it to work for other areas of the registry
+    $regex = '^HKLM:\\([a-zA-Z0-9\s_@\-\^!#.\:\/\$%&+={}\[\]\\*])+$'
```
Comment (wmmatt) [inline Function.Registry-Helpers.ps1:120]: Tested this out a bunch and I couldn't break it so should be good. I saw your note on only HKLM and I can't really think of a time where you really need another registry path other than like HKU but even that is super niche so this should be fine.

**Review by wmmatt — APPROVED** [review]
I am putting approved because I've gone through this and haven't been able to cripple my test VM but just a reminder that remove anything functions are VERY sensitive, and registry removals are even more sensitive. I get that's part of the justification for making the process repeatable but you get me...please test test test before in full prod.

**Pull request merged**